ui/pages/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -134,14 +134,22 @@ const mapStateToProps = (state, ownProps) => {
   const { txParams = {}, id: transactionId, type } = txData;
   const txId = transactionId || paramsTransactionId;
   const transaction = getUnapprovedTransaction(state, txId);
+  const finalTxParams = (transaction && transaction.txParams) || txParams;
+
+  if (!Object.keys(finalTxParams).length) {
```

Comment:
The intent of these `mapStateToProps` functions is to return an object with necessary state extracted but ideally in a consistent format.

But if we add branches that return nothing, then there is less consistency for the component as it has to assume every single property could be `undefined`.

As with the component, would it be cleaner to always return the full state object here with all the applicable properties, some of which may always be present, but if we're the situation where no state is needed, then does that imply the parent component should never have rendered this component to begin with?